ksql-engine/src/main/java/io/confluent/ksql/metrics/Counter.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package io.confluent.ksql.metrics;
+
+import org.apache.kafka.common.metrics.KafkaMetric;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.metrics.Sensor;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+class Counter<R> {
+
+  private final String topic;
+  Map<String, SensorMetric<R>> sensors = new HashMap<>();
+
+  Counter(String topic, final Map<String, SensorMetric<R>> sensors) {
+    this.topic = topic.toLowerCase();
+    this.sensors = sensors;
+  }
+
+  void increment(R record) {
+    sensors.values().forEach(v -> v.record(record));
+  }
+
+  public void close(Metrics metrics) {
+    sensors.values().forEach(v -> v.close(metrics));
+  }
+
+  public boolean isTopic(String topic) {
+    return this.topic.equals(topic);
+  }
+
+  public String statsAsString(boolean verbose) {
```

Comment:
package private? also still wondering why we return stats as a string and not as a `Stats` object?

Reply:
the alternative is to return Stats that has a toString to format results for the cli

Reply:
i guess the thing is will we want to use this in a way that it makes sense to have an object rather than just a string? I guess my preference is to use domain types rather than strings etc everywhere as it is easier to comprehend the code, i.e., a string is a string is a string...

Reply:
still needs to be a string at the end of this exec path, but I get what you mean, I'll pass back the collection the run a formatter thingy over at the top level.